packages/wallets/src/components/ModalProvider/ModalProvider.tsx

```diff
@@ -50,9 +51,12 @@ const ModalProvider = ({ children }: React.PropsWithChildren<unknown>) => {
         };
     };
 
-    const show = (ModalContent: React.ReactNode, options?: TModalShowOptions) => {
+    const show = (ModalContent: React.ReactNode, options?: TModalOptions) => {
         setContent(ModalContent);
-        setCustomRootRef(options?.rootRef?.current ? options?.rootRef : null);
+        setModalOptions({
+            ...modalOptions,
+            ...options,
+        });
```

Comment:
Responsive modal can be mounted to desktop root by:
```
show(<YourModal />, {
   defaultRootId: 'wallets_modal_root'
})
```
<img width="1840" alt="Screenshot 2023-11-06 at 5 00 29 PM" src="https://github.com/binary-com/deriv-app/assets/103016120/0753ca66-6683-4ea4-8e1d-ebe3b3e69255">